packages/@aws-cdk/aws-amplify-alpha/test/integ.app-asset-deployment.ts

```diff
@@ -1,25 +1,28 @@
-// import * as path from 'path';
-// import { Asset } from '@aws-cdk/aws-s3-assets';
+import * as path from 'path';
 import { App, Stack, StackProps } from 'aws-cdk-lib';
 import { Construct } from 'constructs';
 import * as amplify from '../lib';
+import { Asset } from 'aws-cdk-lib/aws-s3-assets';
+import { IntegTest } from '@aws-cdk/integ-tests-alpha';
 
 class TestStack extends Stack {
   constructor(scope: Construct, id: string, props?: StackProps) {
     super(scope, id, props);
 
-    // FIXME: currently this is only released as part of amplify-alpha, which cannot be tested here
-    // and the asset hash is different in alpha
+    // Deploying this integ test would create an Amplify app with branch name 'main'
+    // Re-deploying the stack again after updating the asset file would update the Amplify app.
     //
-    // const asset = new Asset(this, 'SampleAsset', {
-    //   path: path.join(__dirname, './test-asset'),
-    // });
+    const asset = new Asset(this, 'SampleAsset', {
+      path: path.join(__dirname, './test-asset'),
+    });
 
-    new amplify.App(this, 'App', {});
-    // amplifyApp.addBranch('main', { asset });
+    const amplifyApp = new amplify.App(this, 'App', {});
+    amplifyApp.addBranch('main', { asset });
   }
 }
 
 const app = new App();
-new TestStack(app, 'cdk-amplify-app-asset-deployment');
-app.synth();
+const stack = new TestStack(app, 'cdk-amplify-app-asset-deployment');
+new IntegTest(app, 'cdk-amplify-app-integ-test', {
+  testCases: [stack],
+});
```

Comment:
There are no assertions here -- can you document why you think deployment of the stack is sufficient to test the two custom resources?

Reply:
Did add some comments explaining the working of custom resources as expected with stack deployment.